### PR TITLE
refactor: update deployment documentation for local development using start-dev.sh

### PR DESCRIPTION
**Pull request by Aravinda-HWK**
## Summary

This PR is to remove the `docker-compose.ym`l file and `start-docker.sh` files for local development because the developers only use `start-dev.sh` file and it does not use docker compose for run the system

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [x] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Changes Made

- Update the documentation
- Remove docker-compose file
- Remove start-docker.sh file  

## Testing

- [x] I have tested this change locally
- [ ] I have added unit tests for new functionality
- [ ] I have tested edge cases
- [ ] All existing tests pass

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

### CODE REVIEW

File: docs/DEPLOYMENT.md [hunk 1]
```diff
@@ -98,54 +97,40 @@ The backend requires a `services.json` file to manage outbound connections to ex
...
-This is the reference deployment path currently implemented in this repository.
+This is the supported local development path in this repository.
 
 ### 5.1 What is provided
 
-- Root `docker-compose.yml`
-- Wrapper script `start-docker.sh`
-- Profiles for optional IDP/DB-managed components
-- Persistent named volumes for DBs and backend uploads
+- Root `start-dev.sh` dev runner
```
[the remark this answers is not in the document]
Comment (Aravinda-HWK) [reply docs/DEPLOYMENT.md:115]: No need to update this in the documentation.

**Review by sthanikan2000 — APPROVED** [review]

**Review by ginaxu1 — APPROVED** [review]
